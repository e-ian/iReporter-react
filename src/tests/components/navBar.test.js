import { shallow } from "enzyme";
import React from "react";
import NavbarComponent from "../../components/navBarComponent";

describe("create intervention component success tests", () => {
  const props = {
    onclick: jest.fn(),
    logout: jest.fn(),
    history: {push:jest.fn()}
  };
  let wrapper;
  let instance;

  it("should render without crashing", () => {
    const wrapper = shallow(<NavbarComponent {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
  it("should handle onclick event", () => {
    const wrapper = shallow(<NavbarComponent {...props} />);
    instance = wrapper.instance();
    instance.logout();
    expect(props.logout).toBeCalledTimes(0);
  });
});
