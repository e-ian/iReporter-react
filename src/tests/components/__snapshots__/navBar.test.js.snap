// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create intervention component success tests should render without crashing 1`] = `ShallowWrapper {}`;
